core/shared/src/main/scala-3/zio/prelude/Newtype.scala

```diff
@@ -5,14 +5,38 @@ import zio.NonEmptyChunk
 import scala.quoted.*
 import zio.prelude.ConsoleUtils.*
 
-abstract class Newtype[A] {
+abstract class NewtypeCustom[A] {
```

Comment:
Do we want to make it private? Because we don't have an `NewtypeCustom` for Scala 2 flavor, right? By making this public, we're introducing an asymmetry between Scala 2 and Scala 3 flavors of ZIO Prelude.

Reply:
I think the goal of this PR is to expose this functionality for end users to define their own custom compile time assertions (without forcing the additional complexity of doing so on normal users) so I think this needs to be public.

My question would be whether we can expose similar functionality in Scala 2? If not I don't think it is the end of the world if this is a Scala 3 only feature if we don't think there is a feasible way to support an equivalent on Scala 2.

Reply:
For scala 2 no extra ceremony should be necessary https://github.com/zio/zio-prelude/pull/793/files#diff-cb2c8d5aab7fb0cedcd2c4d8dbfad7777e0a9761ea3f7a5f09ad205695825880R205-R210

https://github.com/zio/zio-prelude/pull/793/files#diff-109e476632c38a367dd0aa6740e67ad6801296fbed9592f638360a41d2334922R67-R73

Reply:
@mschuwalow Interestingly, I think I've discovered an issue with `SubtypeCuston`. It doesn't seem to behave like a `Subtype` but like a `Newtype`

See:
- https://github.com/zio/zio-prelude/pull/1327
- https://github.com/zio/zio-prelude/actions/runs/9316877282/job/25645999588?pr=1327